pkg/glanceapi/deployment.go

```diff
@@ -100,7 +98,7 @@ func Deployment(
 
 	deployment := &appsv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      fmt.Sprintf("%s-api", instance.Name),
+			Name:      instance.Name,
```

Comment:
I'm assuming this is already `%s-api` because you moved the logic in the controller.